contracts/pcv/ERC20HoldingPCVDeposit.sol

```diff
@@ -0,0 +1,110 @@
+    IERC20 public immutable override token;
+
+    /// @notice address of VOLT on mainnet
+    address private constant MAINNET_VOLT =
+        0x559eBC30b0E58a45Cc9fF573f77EF1e5eb1b3E18;
+
+    /// @notice address of VOLT on arbitrum
+    address private constant ARBITRUM_VOLT =
+        0x6Ba6f18a290Cd55cf1B00be2bEc5c954cb29fAc5;
+
+    constructor(address _core, IERC20 _token) CoreRef(_core) {
+        require(
+            address(_token) != MAINNET_VOLT && address(_token) != ARBITRUM_VOLT,
+            "VOLT not supported"
+        );
+        token = _token;
+    }
+
+    /// @notice Empty receive function to receive ETH
+    receive() external payable {}
+
+    ///////   READ-ONLY Methods /////////////
+
+    /// @notice returns total balance of PCV in the deposit
+    function balance() public view override returns (uint256) {
+        return token.balanceOf(address(this));
+    }
+
+    /// @notice returns the resistant balance and FEI in the deposit
+    function resistantBalanceAndVolt()
+        public
+        view
+        override
+        returns (uint256, uint256)
+    {
+        return (balance(), 0);
+    }
+
+    /// @notice display the related token of the balance reported
+    function balanceReportedIn() public view override returns (address) {
+        return address(token);
+    }
+
+    /// @notice No-op deposit
+    function deposit()
+        external
+        override(IERC20HoldingPCVDeposit, IPCVDeposit)
+        whenNotPaused
+    {
+        emit Deposit(msg.sender, balance());
+    }
+
+    /// @notice Withdraw underlying
+    /// @param amountUnderlying of tokens withdrawn
+    /// @param to the address to send PCV to
+    function withdraw(address to, uint256 amountUnderlying)
+        external
+        override(IERC20HoldingPCVDeposit, IPCVDeposit)
+        hasAnyOfTwoRoles(TribeRoles.GOVERNOR, TribeRoles.PCV_CONTROLLER)
+    {
+        token.safeTransfer(to, amountUnderlying);
+        emit Withdrawal(msg.sender, to, amountUnderlying);
+    }
+
+    /// @notice Withdraw all of underlying
+    /// @param to the address to send PCV to
+    function withdrawAll(address to)
+        external
+        hasAnyOfTwoRoles(TribeRoles.GOVERNOR, TribeRoles.PCV_CONTROLLER)
+    {
+        uint256 amountUnderlying = token.balanceOf(address(this));
+        token.safeTransfer(to, amountUnderlying);
+        emit Withdrawal(msg.sender, to, amountUnderlying);
+    }
+
+    /// @notice Wraps all ETH held by the contract to WETH. Permissionless, anyone can call it
+    function wrapETH() public {
+        uint256 ethBalance = address(this).balance;
+
+        if (ethBalance != 0) {
+            if (block.chainid == 1) {
+                Constants.WETH.deposit{value: ethBalance}();
+            } else if (block.chainid == 42161) {
+                Constants.ARBITRUM_WETH.deposit{value: ethBalance}();
+            } else {
+                revert("Can only wrap eth on mainnet and arbitrum");
+            }
+        }
+    }
```

Comment:
For this we could store the WETH addresses for different chains in some sort of mapping, and pass the `chainId` instead to this function, that way if we want to deploy on say optimism we wouldn't have to redeploy this contract, wdyt?

Reply:
we can just add the optimism param in here, that will be cheaper than doing an SLOAD to figure out what network we're on